### ISSUE TITLE
Blazor Web App / Auto Mode / Counter button not responsive if switch to that page quickly

### ISSUE THREAD

**Issue by ToolmakerSteve** [dotnet/aspnetcore#58860]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

New Project / Blazor Web App / Interactive Render Mode = Auto.

Launch under debugger. Quickly switch to Counter page, and quickly press Counter.

It does not respond (WASM not yet loaded).

### Expected Behavior

Counter is always responsive.

(Stay in Server Render Mode, until a page switch occurs AND WASM is ready.)

IMHO, this should be the default behavior: "Keep simple cases simple".

To put it another way: what possible benefit is there to switching to client rendering **when Blazor knows perfectly well that cannot work yet**!

### Steps To Reproduce

See bug description.

### .NET Version

latest 8.0

### Anything else?

This bug appears to be "by design", and probably covered by https://github.com/dotnet/aspnetcore/issues/53799,
as well as similar issues.

Nevertheless, even the simplest app reveals a problem with the current logic for switching between server and client rendering.

Please don't lose sight of this fact.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.

**Comment by ToolmakerSteve** [comment]
@mkArtakMSFT - I am fairly sure that you are describing a different scenario than what I described.

To be clear: I did not suggest **dynamically** switching. I am pointing out that Blazor makes a decision to switch when user switches page. At that decision point, isn't it easy to determine whether WebAssembly is ready to start up?

Currently, there does not appear to be such a check. Blazor simply assumes the first page switch, it can switch to WebAsm. Even if WebAssembly is still being downloaded. According to the tests I did.

My suggestion, in this situation, is to wait until the NEXT page change.
